pkg/ast/ast.go

```diff
@@ -221,7 +221,9 @@ func (slf *PrefixCondition) String() string {
 }
 func (slf *PrefixCondition) GetTypes() []*Identifier {
 	out := []*Identifier{}
-	out = append(out, slf.Right.GetTypes()...)
+	if slf.Right != nil {
```

Comment:
any comparison to nil needs to use IsNilInterface()
```suggestion
	if !types.IsNilInterface(slf.Right) {
```

need to use this same pattern when comparing interfaces